vm.go

```diff
@@ -102,13 +102,19 @@ func (l *State) equalObjects(t1, t2 value) bool {
 	case *userData:
 		if t1 == t2 {
 			return true
+		} else if t2, ok := t2.(*userData); ok && t2 != nil {
+			tm = l.equalTagMethod(t1.metaTable, t2.metaTable, tmEq)
+		} else if !ok || t2 == nil {
+			return false
```

Comment:
Sorry, I missed the fall-through to the tag method call. You can simply remove the `return false` in the code suggested above. This then falls through to `return tm != nil && ...`, which is equivalent your `else if` clause. So the code for this case is:
```go
	case *userData:
		if t1 == t2 {
			return true
		} else if t2, ok := t2.(*userData); ok {
			tm = l.equalTagMethod(t1.metaTable, t2.metaTable, tmEq)
		}
```